### DISCUSSION TITLE
prepared statement \"s4\" already exists / Cant connect (USING SUPABASE)

### DISCUSSION THREAD

**Discussion by 0w9** [prisma/prisma#14958] [Q&A]
### Bug description

Hello. I have come across such an issue often lately. Here is a description:

I am using Supabase, and the exact given URIs. 
I keep getting an error, involving "prepared statement \"s4\" already exists" (Sometimes also prepared statement \"s0\" already exists"). 

Other error in the same context: "ERROR	PrismaClientInitializationError: Error querying the database: DB error: FATAL: pgbouncer cannot connect to server"

After restarting It often gets solved for one interaction, and then appears again.

MY URI: postgresql://USERNAME:PASSWORD@db.URL.supabase.co:5432/postgres?schema=public
(I have tried to add `&pgboucer=true`, etc.)

I would love to hear why this might happen. I SUSPECT IT TO BE POOLING. I tried other ports too.

Thanks in advance

### Expected behavior

I expect the connection to work and write to the DB.

### Prisma information
```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = "postgresql://USERNAME:PASSWORD@db.URL.supabase.co:5432/postgres?schema=public"
}

model contracts {
  id      Int      @id @default(autoincrement()) @db.SmallInt
  name    String?  @default("Smart Contract ")
  type    Int?     @db.SmallInt
  address String?  @db.VarChar
  creator Int
  website String[]
  users   users    @relation(fields: [creator], references: [id], onDelete: NoAction, onUpdate: NoAction)
}

model tokens {
  unhashed_token Int     @default(autoincrement()) @db.SmallInt
  token          String? @db.VarChar
  id             Int     @default(autoincrement()) @db.SmallInt
  user           Int     @id @db.SmallInt
  users          users   @relation(fields: [user], references: [id], onDelete: NoAction, onUpdate: NoAction)
}

model users {
  id               Int         @id @db.SmallInt
  wallet_pubKey    String?     @default("") @db.VarChar(524)
  wallet_secretKey String?
  discord_id       String?
  username         String
  password         String
  avatar_url       String
  email            String
  solana           String?     @db.VarChar(524)
  usertype         Int?
  role             String?
  twitter          String?     @db.VarChar
  contracts        contracts[]
  tokens           tokens?
}
```

### Environment & setup

- OS: MacOS (m1 chip) and Vercel
- Database: PostgreSQL
- Node.js version: v16.15.0

### Prisma Version

```
Environment variables loaded from prisma/.env
prisma                  : 4.2.1
@prisma/client          : 3.15.2
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine 2920a97877e12e055c1333079b8d19cee7f33826 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli 2920a97877e12e055c1333079b8d19cee7f33826 (at node_modules/@prisma/engines/migration-engine-darwin-arm64)
Introspection Engine    : introspection-core 2920a97877e12e055c1333079b8d19cee7f33826 (at node_modules/@prisma/engines/introspection-engine-darwin-arm64)
Format Binary           : prisma-fmt 2920a97877e12e055c1333079b8d19cee7f33826 (at node_modules/@prisma/engines/prisma-fmt-darwin-arm64)
Default Engines Hash    : 2920a97877e12e055c1333079b8d19cee7f33826
Studio                  : 0.469.0
✨  Done in 1.36s.
```

**Comment by janpio** [comment]
Are you using the pooled connection string from Supabase or the direct one?
(The `5432` as port indicates you are using the direct one - but the other error message your are sharing, `Error querying the database: DB error: FATAL: pgbouncer cannot connect to server` point in the other direction)

Using `&pgbouncer=true` in the connection string should only be necessary (and useful) when you are actually using the connection pooled connection.

> Other error in the same context: "ERROR PrismaClientInitializationError: Error querying the database: DB error: FATAL: pgbouncer cannot connect to server"

This is an error message from Supabase's pgbouncer that we can not influence. This one you will need to report to Supabase directly and let them figure out what goes wrong in communication between their pgbouncer and database.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
